phnt/include/ntldr.h

```diff
@@ -662,12 +662,12 @@ typedef struct _PS_SYSTEM_DLL_INIT_BLOCK
             ULONG Reserved : 31;
         };
     };
-    PS_MITIGATION_OPTIONS_MAP MitigationOptionsMap;
-    ULONG_PTR CfgBitMap;
+    PS_MITIGATION_OPTIONS_MAP MitigationOptionsMap; // current size since 20H1
+    ULONG_PTR CfgBitMap; // current offset since 20H1
```

Comment:
Which field is since 20H1?

Reply:
That depends on what we want to mark as "since". `PS_MITIGATION_OPTIONS_MAP` contained a two element array before 20H1, now it's three. So `MitigationOptionsMap` used to start at the same offset but was smaller. Thus, `current size since 20H1`. And `CfgBitMap` used to start earlier, thus, `current offset since 20H1`.

Reply:
I think two separate structures are needed with a V1/2 suffix since this and the below MitigationAuditOptionsMap for RS3 are incompatible with later versions?

Reply:
Yes, sure. If we want to support as many fields as possible on as many OS versions as possible, three revisions seem like the minimum (since MS broke offsets twice). One structure for 8-RS1, another for RS2-19H2, and the newest for 20H1 and above. I can add them to `ntldr.h` and then hide as an implementation detail in `PhGetProcessSystemDllInitBlock` by making sure it always converts them to the latest format before returning. Does that sound good?

Reply:
Yes